### PR TITLE
created function to return the the current winner and functionality to check draw state of baord and used LLVM clang-format

### PR DESCRIPTION
**Pull request by Arslan909**
made a function which check the board state and return the winner as char also added functionality to check the draw state

### CODE REVIEW

**Review by shasherazi — APPROVED** [review]
Thek hi hoga :shrug:

**Comment by Arslan909** [comment]
> Thek hi hoga shrug

File: include/board.c [hunk 1]
```diff
@@ -40,4 +40,48 @@ void turn_change(struct Board *board) {
...
+      if (board->board[row][col] != 'X' && board->board[row][col] != 'O') {
+        return 0;
+      }
+    }
+  }
+  return 1;
+}
+
+// 0: no winner, 1: player wins
+int check_win(struct Board *board) {
```
Comment (shasherazi) [inline include/board.c]: `board->turn` is a `char`, not an `int`.

File: include/board.c [hunk 2]
```diff
@@ -40,4 +40,48 @@ void turn_change(struct Board *board) {
...
+    for (int col = 0; col < 3; col++) {
+      if (board->board[row][col] != 'X' && board->board[row][col] != 'O') {
+        return 0;
+      }
+    }
+  }
+  return 1;
+}
+
+// 0: no winner, 1: player wins
```
Comment (shasherazi) [inline include/board.c]: modify comment according to the function working

**Review by shasherazi — CHANGES_REQUESTED** [review]

**Review by shasherazi — APPROVED** [review]

**Pull request merged**